Added XmlExtensions Unit Tests.

michelvosje opened the pull request:
Implemented the XmlExtensionsUnitTests class for ticket #14.

Commit:
Added XmlExtensions Unit Tests.

Commit:
Merge branch 'master' of https://github.com/Teodor92/MoreDotNet into xmlextensions_tests

Teodor92 commented on Source/MoreDotNet.Test/Extensions/Common/XmlExtensions/XmlExtensionsTests.cs:
```diff
@@ -0,0 +1,44 @@
+        {
+            object testObject = new object();
+
+            Assert.NotNull(testObject.XmlSerialize());
```
This just checks if the returned string is not null, but is it truly correctly serialized? An additional check will make the test better.

Teodor92 commented on Source/MoreDotNet.Test/Extensions/Common/XmlExtensions/XmlExtensionsTests.cs:
```diff
@@ -0,0 +1,44 @@
+            object testObject = new object();
+            string testSerializedObject = testObject.XmlSerialize();
+
+            Assert.NotNull(testSerializedObject.XmlDeserialize<object>());
```
Same as above - it necessary to check if the deserialization is correct.

Teodor92 commented on Source/MoreDotNet.Test/MoreDotNet.Tests.csproj:
```diff
@@ -40,9 +40,13 @@
   <ItemGroup>
+    <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <Private>False</Private>
+    </Reference>
```
Please remove the reference to Microsoft.VisualStudio.QualityTools.UnitTestFramework.

Teodor92 reviewed (changes requested):
Have in mind that there are a few test cases missing. For example:
- What happens if you try to deserialize to an incorrect type?

Anyway if you fix the few remarks I made in the comments, I will merge the PR :)

michelvosje commented:
Yes you are right. Got a little bit short on time. Will fix it tomorrow.

Commit:
Fixed issues in XmlExtensionsTests.
+ Validates whether serialized objects are in valid XML format.
+ Validates whether objects are correctly deserialized.

michelvosje commented:
I fixed the issues. What now?

Teodor92 commented:
Merged!

michelvosje commented:
Doing even small tickets like these give me more experience with git, Github and how to contribute to open projects.
Ty.
